cli/src/lib/compiler/compile.js

```diff
@@ -10,14 +10,11 @@ const {
     createAppEntrypointWrapper,
     createPluginEntrypointWrapper,
 } = require('./entrypoints.js')
-const {
-    extensionPattern,
-    normalizeExtension,
-} = require('./extensionHelpers.js')
+const { extensionPattern } = require('./extensionHelpers.js')
 
 const watchFiles = ({ inputDir, outputDir, processFileCallback, watch }) => {
     const compileFile = async (source) => {
-        const relative = normalizeExtension(path.relative(inputDir, source))
+        const relative = path.relative(inputDir, source)
```

Comment:
This was what broke JSX files in platform apps/libs: as an example, consider `index.jsx` has `import App from 'App.jsx'`. At compile time, `App.jsx` gets converted to `App.js`, but the import text in `index` is still from `App.jsx` (it doesn't get modified despite the filename change), and then an error is thrown that says "Can't find `App.jsx`"